server/apiCalls/ugRestApi.js

```diff
@@ -0,0 +1,150 @@
+  const employee = {}
+  if (assistants.length) {
+    const assistantGroup = groupsAlongWithMembers.find(x => x.name === assistants[0])
+    if (assistantGroup) {
+      const flatArrWithHtmlStr = createPersonHtml(assistantGroup.members)
+      employee.assistants = flatArrWithHtmlStr
+    }
+  }
+  if (teachers.length) {
+    const teachersGroup = groupsAlongWithMembers.find(x => x.name === teachers[0])
+    if (teachersGroup) {
+      const flatArrWithHtmlStr = createPersonHtml(teachersGroup.members)
+      employee.teachers = flatArrWithHtmlStr
+    }
+  }
+  if (examiners.length) {
+    const examinersGroup = groupsAlongWithMembers.find(x => x.name === examiners[0])
+    if (examinersGroup) {
+      const flatArrWithHtmlStr = createPersonHtml(examinersGroup.members)
+      employee.examiners = flatArrWithHtmlStr
+    }
+  }
+  if (responsibles.length) {
+    const responsiblesGroup = groupsAlongWithMembers.find(x => x.name === responsibles[0])
+    if (responsiblesGroup) {
+      const flatArrWithHtmlStr = createPersonHtml(responsiblesGroup.members)
+      employee.responsibles = flatArrWithHtmlStr
+    }
+  }
+  return employee
+}
+
+/**
+ * This will first prepare filter query then pass that query to ug rest api and then return groups data along with members.
+ * @param assistants Assistants group name
+ * @param teachers Teachers group name
+ * @param examiners Examiners group name
+ * @param responsibles Responsibles group name
+ * @returns Will return groups along with members from ug rest api.
+ */
+async function getAllGroupsAlongWithMembersRelatedToCourse(assistants, teachers, examiners, responsibles) {
+  const { url, key } = serverConfig.ugRestApiURL
+  const { authTokenURL, authClientId, authClientSecret } = serverConfig.ugAuth
+  const ugConnectionProperties = {
+    ugTokenURL: authTokenURL,
+    clientId: authClientId,
+    clientSecret: authClientSecret,
+    ugURL: url,
+    subscriptionKey: key,
+  }
+  ugRestApiHelper.initConnectionProperties(ugConnectionProperties)
+  const filterData = []
+  if (assistants.length) {
+    filterData.push(assistants[0])
+  }
+  if (teachers.length) {
+    filterData.push(teachers[0])
+  }
+  if (examiners.length) {
+    filterData.push(examiners[0])
+  }
+  if (responsibles.length) {
+    filterData.push(responsibles[0])
+  }
+  log.info('Started fetching data of groups along with members from UG Rest Api at ', getCurrentDateTime())
+  log.info('Fetching groups along with members')
+  log.info('Using Filter Query Data', { filterData })
+  log.info('Using Filter Operator:', 'in')
+  const groupDetails = await ugRestApiHelper.getUGGroups('name', 'in', filterData, true)
+  log.info('Successfully fetched data of groups along with members from UG Rest Api at ', getCurrentDateTime())
+  log.info('Successfully fetched groups data along with members')
+  log.info('Filter Query Data Used', { filterData })
+  log.info('Filter Operator Used :', 'in')
```

Comment:
- Pipeline team have asked to use as little as possible logs, so it's better reduce it to 2 logs, 1 before and 1 after with, f.e.,` log.info('some text', {courseCode, semester, time: getCurrentDateTime()})` and here it's unclear a bit for which courseCode and semester it's fetched, so it will be hard to search in graycloud logs

Reply:
Remove extra logs.